x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/step_define_package_policy.tsx

```diff
@@ -208,9 +208,9 @@ export const StepDefinePackagePolicy: React.FunctionComponent<{
 
           {/* Required vars */}
           {requiredVars.map((varDef) => {
-            const { name: varName, type: varType } = varDef;
-            if (!packagePolicy.vars || !packagePolicy.vars[varName]) return null;
-            const value = packagePolicy.vars[varName].value;
+            const { name: varName, type: varType, default: defaultValue } = varDef;
+            const value = packagePolicy.vars?.[varName]?.value ?? defaultValue;
```

Comment:
Should we populate `packagePolicy.vars?.[varName]?.value` with the default value here instead of displaying the default value, I think if the user do not change the value it will not be saved no? (not tested locally)

Reply:
I just pushed up https://github.com/elastic/kibana/pull/132068/commits/fede1404525de2ba37b71d4880c32781646c3236 which should address this. We now create the `vars[varName]` object entry on `packagePolicy` and set the default value if necessary, and the values/vars persist as expected.

![image](https://user-images.githubusercontent.com/6766512/168080645-f182f452-8142-4e87-a557-055cc3e0b320.png)